src/executor/contract.rs

```diff
@@ -150,6 +153,7 @@ impl AotContractExecutor {
             used_function_ids.insert(entry.function_idx as u64);
         }
 
+        let no_eq_solver = sierra_version.minor >= 4;
```

Comment:
i think we should check the major too

Reply:
+1, probably with something like this
```suggestion
        let no_eq_solver = match sierra_version.major.cmp(&1) {
            Ordering::Less => false,
            Ordering::Eq => sierra_version.minor >= 4,
            Ordering::Greater => true,
        };
```

Reply:
Done